### PR TITLE
Event classes and specialised channels

### PR DESCRIPTION
**Pull request by Andrea-Campanella**

### CODE REVIEW

File: pkg/southbound/synchronizer/factory.go [hunk 1]
```diff
@@ -27,23 +27,23 @@ import (
...
-	deviceStore *topocache.DeviceStore, topoChannel <-chan events.Event) {
+	deviceStore *topocache.DeviceStore, topoChannel <-chan events.TopoEvent) {
 
 	for topoEvent := range topoChannel {
-		deviceName := topoEvent.Subject()
-		if topoEvent.EventType() == events.EventTypeTopoCache {
-			if !listener.CheckListener(deviceName) &&
-				topoEvent.Value(events.Connect) == "true" {
+		deviceName := events.Event(topoEvent).Subject()
+		log.Println("type event", events.Event(topoEvent).EventType(), "constant", events.EventTypeTopoCache)
```
Comment (SeanCondon) [inline pkg/southbound/synchronizer/factory.go]: can reduce the printing here

File: pkg/southbound/synchronizer/factory.go [hunk 2]
```diff
@@ -27,23 +27,23 @@ import (
...
+		log.Println("equals ", events.Event(topoEvent).EventType() == events.EventTypeTopoCache)
+		if events.Event(topoEvent).EventType() == events.EventTypeTopoCache {
```
[the remark this answers is not in the document]
Comment (Andrea-Campanella) [reply pkg/southbound/synchronizer/factory.go]: done. forgot them in due to testing

File: pkg/southbound/synchronizer/factory.go (hunk 1 above)
Comment (Andrea-Campanella) [reply pkg/southbound/synchronizer/factory.go]: done

File: pkg/manager/setconfig.go [hunk 3]
```diff
@@ -89,10 +89,7 @@ func (m *Manager) SetNetworkConfig(target string, configName store.ConfigName, u
 	// 		the (YANG) models for the device
 	//  2) Do a precheck that the device is reachable
 	//  3) Check that the caller is authorized to make the change
-	eventValues := make(map[string]string)
-	eventValues[events.ChangeID] = store.B64(configChange.ID)
-	eventValues[events.Committed] = "true"
-	m.ChangesChannel <- events.CreateEvent(deviceConfig.Device,
-		events.EventTypeConfiguration, eventValues)
+	m.ChangesChannel <- events.ConfigEvent(events.CreateConfigEvent(deviceConfig.Device,
```
Comment (SeanCondon) [inline pkg/manager/setconfig.go]: Is this cast needed?

**Pull request merged**